### PR TITLE
introduce a csv printer

### PR DESCRIPTION
**Pull request by thheinel**

### CODE REVIEW

File: pkg/genericcli/printers/csv.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package printers
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"io"
+
+	"github.com/jszwec/csvutil"
```
Comment (majst01) [inline pkg/genericcli/printers/csv.go:9]: you must run `go mod tidy`

Comment (thheinel) [reply pkg/genericcli/printers/csv.go:9]: thanks i forgot :-(

File: pkg/genericcli/printers/csv.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+type CSVPrinterConfig struct {
+	// AutoHeader will generate headers during print, default is go standard ("false")
+	AutoHeader bool
+	// Delimiter the char to separate the columns, default is ";"
+	Delimiter rune
+}
+
+func NewCSVPrinter(config *CSVPrinterConfig) *CSVPrinter {
+	const defaultDelimiter = ';'
```
Comment (majst01) [inline pkg/genericcli/printers/csv.go:25]: const should be toplevel, not inside a func

File: pkg/genericcli/printers/csv.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		return err
+	}
+
+	w.Flush()
+
+	if err := w.Error(); err != nil {
+		return err
+	}
+
+	fmt.Fprintf(cp.out, "%s\n", buf.String())
```
Comment (majst01) [inline pkg/genericcli/printers/csv.go:62]: I can't see where the buffer is filled ??

**Review by majst01 — CHANGES_REQUESTED** [review]

File: pkg/genericcli/printers/csv.go (hunk 2 above)
Comment (thheinel) [reply pkg/genericcli/printers/csv.go]: it's now toplevel

File: pkg/genericcli/printers/csv.go (hunk 3 above)
Comment (thheinel) [reply pkg/genericcli/printers/csv.go]: nor can I. removed unnecessary code

File: pkg/genericcli/printers/csv.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	err := enc.Encode(data)
+	if err != nil {
+		return err
+	}
+
+	w.Flush()
+
+	if err != nil {
+		return err
+	}
```
Comment (majst01) [inline pkg/genericcli/printers/csv.go:56]: err is not set after w.Flush, can be removed

File: pkg/genericcli/printers/csv.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	if config.Delimiter == 0 {
+		config.Delimiter = defaultDelimiter
+	}
+
+	return &CSVPrinter{
+		c: config,
+	}
+}
+
+func (cp *CSVPrinter) WithOut(out io.Writer) *CSVPrinter {
```
Comment (majst01) [inline pkg/genericcli/printers/csv.go:34]: Why do you have a Option for setting the Writer, but a Config for header and delimiter, seems a two fold approach. I think either config or options for all configurable settings.

Comment (thheinel) [reply pkg/genericcli/printers/csv.go:47]: edit out as an output option as well

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**